Added canwear var to pets, in order to prevent invisible pets

Wad67 opened the pull request:
What it says on the tin.
All current pets bar gondola can wear collars, & tags.

Put in place to prevent pets from going invisible if they don't have the correct icon state.
Fixes #33142

Wad67 committed:
Added canwear var to pets, in order to prevent putting collars on pets with no relevant icon

What it says on the tin.
All current pets bar gondola can wear collars, & tags.

Put in place to prevent pets from going invisible if they don't have the correct icon state.

Cyberboss commented on code/modules/mob/living/simple_animal/friendly/gondola.dm:
```diff
@@ -25,6 +25,8 @@
 	health = 200
 	del_on_death = TRUE
+	//cannot wear pet collars and tags because nobody has made relevant icons
+	canwear = 0
```
FALSE

Cyberboss commented on code/modules/mob/living/simple_animal/friendly/pet.dm:
```diff
@@ -4,21 +4,23 @@
 	var/obj/item/clothing/neck/petcollar/pcollar = null
 	var/collar = ""
 	var/pettag = ""
+	var/canwear = 1
```
TRUE

lzimann replied:
this var should be named `can_wear_pet_collar` or something similar, canwear is too ambiguous and vague

Wad67 commented:
Will perform the suggested changes, 
Assuming we are using TRUE & FALSE respectively for code readability?

Cyberboss commented:
Yes

Wad67 committed:
Changed var names to their sane counterparts

ShizCalev commented on code/modules/mob/living/simple_animal/friendly/pet.dm:
```diff
@@ -4,21 +4,23 @@
-			name = real_name
-		qdel(P)
-		return
+		if(can_wear_pet_items)
```
Unnest this.

ShizCalev commented on code/modules/mob/living/simple_animal/friendly/pet.dm:
```diff
@@ -4,21 +4,23 @@
-			name = real_name
-		qdel(P)
-		return
+		if(can_wear_pet_items)
```
Unnest this.

Wad67 replied:
Not immediately obvious what you want un-nested , are you referring to if(P.tagname) ?
Should only apply a name if the pet can receive a collar.

AnturK commented:
If this is only meant to prevent the missing icons instead of being editable just check if the states exist ?

Wad67 commented:
That could most certainly be done in lieu of applying a var. 
If I can get that working, I'll update this pr.

Wad67 committed:
Set map to runtimestation, should not affect anything.

Wad67 committed:
Same as previous, should not change a thing

Wad67 committed:
Changed to detection of applicable icon state

Removed the var based system, now automatically detects if the mob attackedby petcollar can accept a collar.

Wad67 commented:
Shit, I think the gitignore did not apply for some reason

Wad67 commented:
Going to close this because I fucked up.
